gitlab-runner-mock/src/api/request.rs

```diff
@@ -138,6 +138,13 @@ impl Respond for JobRequestResponder {
               { "key": "GITLAB_USER_LOGIN", "value": "test", "public": true, "masked": false },
               { "key": "GITLAB_USER_NAME", "value": "Test User", "public": true, "masked": false }
             ]);
+            if !job.variables().is_empty() {
```

Comment:
Hmm with this refactoring the mock job builder should really just push all the default variables into the MockJob itself rather then having it hardcoded in the responder.